### PR TITLE
distro: Add WithIDGenerator option

### PR DESCRIPTION
**Pull request by sergiopastan**
This adds the WithIDGenerator option mentioned in this issue: https://github.com/signalfx/splunk-otel-go/issues/1377

### CODE REVIEW

**Comment by sergiopastan** [comment]
I have read the CLA Document and I hereby sign the CLA

File: distro/config.go [hunk 1]
```diff
@@ -147,3 +148,15 @@ func WithLogger(l logr.Logger) Option {
 		c.Logger = l
 	})
 }
+
+// WithIDGenerator configures the ID Generator used to generate span and trace IDs.
+//
+// The trace.IDGenerator provided should implement the trace.IDGenerator interface
+// with NewIDs and NewSpanID methods.
+//
+// If this option is not provided, a defaultIDGenerator will be used.
```
Comment (pellared) [inline distro/config.go:157]:
```suggestion
// WithIDGenerator configures the generator used to generate span and trace IDs.
// If this option is not provided, the SDK's default ID generator will be used.
```

File: distro/otel_test.go [hunk 2]
```diff
@@ -341,6 +342,20 @@ func TestTracesResource(t *testing.T) {
 	assertResource(t, got.Resource.GetAttributes())
 }
 
+func TestTracesSpans(t *testing.T) {
```
Comment (pellared) [inline distro/otel_test.go:345]:
```suggestion
func TestWithIDGenerator(t *testing.T) {
```

**Review by pellared — COMMENTED** [review]
Can you also update the `CHANGELOG.md`?

**Comment by sergiopastan** [comment]
Thank you for your comments @pellared . I already added your suggestions and the changelog

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
> do I have to create a signed commit?

Yes. This may help you: https://superuser.com/questions/397149/can-you-gpg-sign-old-commits

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Add
+
+- Add the `WithIDGenerator` option to
+  `github.com/signalfx/splunk-otel-go/distro`. (#2634)
```
Comment (MrAlias) [inline CHANGELOG.md:14]:
```suggestion
- Add the `WithIDGenerator` option to `github.com/signalfx/splunk-otel-go/distro`. (#2634)
```

**Review by MrAlias — APPROVED** [review]

**Comment by sergiopastan** [comment]
Hi @pellared I think it is ok right now. Let me know if anything else is needed. Thanks!

**Comment by pellared** [comment]
@sergiopastan All good. Thanks for your contribution 🎉  I will merge once the CI passes.

**Pull request merged**